### PR TITLE
Pbi 742

### PR DESCRIPTION
**Pull request by aeng7**
Implemented Vast 3.0 skipoffset functionality.

### CODE REVIEW

**Comment by aeng7** [comment]
Please do not merge until Mjolnir PR 486 has been merged.

**Comment by aeng7** [comment]
Update to pass seconds rather than ms to AMC via showSkipVideoAdButton

File: js/ad_manager_vast.js [hunk 1]
```diff
@@ -93,12 +104,47 @@ OO.Ads.manager(function(_, $) {
...
+    }, this);
+
+    /**
+     * Checks to see if this ad manager supports a given Vast version.
+     * @private
+     * @method Vast#supportsVersion
+     * @returns {boolean} true if the version is supported by this ad manager, false otherwise
+     */
+    var supportsVersion = _.bind(function(version) {
+      return _.contains(SUPPORTED_VERSIONS, version);
```
Comment (pilievOoyala) [inline js/ad_manager_vast.js]: I believe there is a version 2.1. I remember encountering this in the past and we had parsing errors.

Comment (aeng7) [inline js/ad_manager_vast.js]: Got it, I'll add logic for supported Vast versions to check major versions (and minor if necessary).

File: js/ad_manager_vast.js [hunk 2]
```diff
@@ -305,6 +349,46 @@ OO.Ads.manager(function(_, $) {
     };
 
     /**
+     * Determine if a Vast ad is skippable, and if so, when the skip ad button should be displayed.
+     * Notifies AMC of the result.
+     * @private
+     * @method Vast#calculateSkipAdOffset
+     * @param {object} adWrapper The current Ad's metadata.
+     */
+    var calculateSkipAdOffset = _.bind(function(adWrapper) {
```
Comment (pilievOoyala) [inline js/ad_manager_vast.js]: when it's name starts with calculate, it makes me think it should return a value. Maybe name it initSkipAdOffset?

Comment (aeng7) [inline js/ad_manager_vast.js]: Good point! Will do

File: js/ad_manager_vast.js [hunk 3]
```diff
@@ -740,6 +825,7 @@ OO.Ads.manager(function(_, $) {
      * @private
      * @method Vast#parseLinearAd
      * @param {xml} Xml containing the ad data to be parsed.
+     * @param {number} version The Vast version
```
Comment (pilievOoyala) [inline js/ad_manager_vast.js]: Is this optional? I doesn't look like it being passed into the function ebcause it's only got one param

Comment (aeng7) [inline js/ad_manager_vast.js]: I think this is a remnant of a previous checkin. Will remove. Nice catch!

**Comment by aeng7** [comment]
Included Peter's suggestions.

**Comment by pilievOoyala** [comment]
Looks good.

**Pull request merged**